kse/src/org/kse/gui/actions/SignJarAction.java

```diff
@@ -85,43 +85,88 @@ protected void doAction() {
 			DSignJar dSignJar = new DSignJar(frame, privateKey, keyPairType, alias);
 			dSignJar.setLocationRelativeTo(frame);
 			dSignJar.setVisible(true);
 
+			// check if jar sign dialog was successful
+			if (!dSignJar.isSuccessful()) {
+				return;
+			}
+
 			SignatureType signatureType = dSignJar.getSignatureType();
 			String signatureName = dSignJar.getSignatureName();
-			File inputJarFile = dSignJar.getInputJar();
-			File outputJarFile = dSignJar.getOutputJar();
+			File[] inputJarFile = dSignJar.getInputJar();
+			List<File> outputJarFile = dSignJar.getOutputJar();
 			String tsaUrl = dSignJar.getTimestampingServerUrl();
+			DigestType digestType = dSignJar.getDigestType();
+
+			// start jar signing process
+			DSignJarSigning dSignJarSigning = new DSignJarSigning(frame, inputJarFile, outputJarFile, privateKey, certs,
+					signatureType, signatureName, signer, digestType, tsaUrl, provider);
+			dSignJarSigning.setLocationRelativeTo(frame);
+			dSignJarSigning.startDSignJarSigning();
+			dSignJarSigning.setVisible(true);
 
-			if (signatureType == null) {
+			// check if jar signing was successful
+			if (!dSignJarSigning.isSuccessful()) {
 				return;
 			}
 
-			String signer = KSE.getFullApplicationName();
-
-			DigestType digestType = dSignJar.getDigestType();
-
-			if (inputJarFile.equals(outputJarFile)) {
-				JarSigner.sign(inputJarFile, privateKey, certs, signatureType, signatureName, signer, digestType,
-						tsaUrl, provider);
+			// check if exceptions were caught during jar signing
+			if (dSignJarSigning.getFileExceptions().size() > 0) {
+				String strOK = res.getString("SignJarAction.ButtonOK.message");
+				String strView = res.getString("SignJarAction.ButtonView.message");
+				// set view and ok buttons
+				Object[] buttons = { strView, strOK };
+				// set file count
+				String fileCount = Integer.toString(inputJarFile.length);
+				// set exception count
+				String errorCount = Integer.toString(dSignJarSigning.getFileExceptions().size());
+				// set file and exception count to object array
+				Object[] objCount = { errorCount, fileCount };
+				// set message string
+				String message = MessageFormat.format(res.getString("SignJarAction.SignJarError.message"), objCount);
+				// set option dialog with message, view, and ok buttons
+				int selected = JOptionPane.showOptionDialog(frame, message,
+						res.getString("SignJarAction.SignJar.Title"), JOptionPane.DEFAULT_OPTION,
+						JOptionPane.INFORMATION_MESSAGE, null, buttons, buttons[1]);
+				// if view button pressed show error collection
+				if (selected == 0) {
+					DErrorCollection dError = new DErrorCollection(frame, dSignJarSigning.getFileExceptions());
+					dError.setVisible(true);
+				}
 			} else {
-				JarSigner.sign(inputJarFile, outputJarFile, privateKey, certs, signatureType, signatureName, signer,
-						digestType, tsaUrl, provider);
+				// set file count
+				String fileCount = Integer.toString(inputJarFile.length);
```

Comment:
Hello @kaikramer and @Colbix,

_The same..._
- Is it possible to change the type of `fileCount` from `String` to `Integer`?
_(especially for improve i18n on resources files)_

Could you make a PR?
or would you like that I make one?